libindi/drivers/auxiliary/lpm.cpp

```diff
@@ -0,0 +1,369 @@
+            ResetB[0].s = ISS_OFF;
+            ResetBP.s   = IPS_OK;
+            IDSetSwitch(&ResetBP, nullptr);
+
+            AverageReadingN[0].value = 0;
+            AverageReadingN[1].value = 0;
+            AverageReadingN[2].value = 0;
+            AverageReadingN[3].value = 0;
+
+            sumSQ = 0;
+            count = 0;
+            return true;
+        } else if (!strcmp(name, SaveBP.name))
+        {
+            IUUpdateSwitch(&SaveBP, states, names, n);
+
+            if (SaveB[SAVE_READINGS].s == ISS_ON)
+            {
+                LOGF_INFO("Save readings to %s", filename);
+                if (fp == nullptr) {
+                    openFilePtr();
+                }
+                SaveBP.s = IPS_OK;
+            } else {
+                LOG_INFO("Discard readings");
+                if (fp != nullptr) {
+                    LOG_DEBUG("close file pointer");
+                    fclose(fp);
+                    fp = nullptr;
+                } else {
+                    LOG_WARN("no file open!");
+                }
+                SaveBP.s = IPS_IDLE;
+            }
+
+            IDSetSwitch(&SaveBP, nullptr);
+            return true;
+        }
+
+    }
+
+    return DefaultDevice::ISNewSwitch(dev, name, states, names, n);
+}
+
+void LPM::openFilePtr()
+{
+    LOG_DEBUG("open file pointer");
+    mkdir(RecordFileT[0].text,S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+    fp = fopen(filename, "a");
+}
+
+bool LPM::getReadings()
+{
+    const char *cmd = "V#";
+    char res[32] = {0};
+    int nbytes_written = 0;
+    int nbytes_read = 0;
+
+    tty_write_string(PortFD, cmd, &nbytes_written);
+    if (tty_read_section(PortFD, res, '#', 60000, &nbytes_read) == TTY_OK)
+    {
+        LOGF_DEBUG("RES (%s)", res);
+        float mpsas;
+        int rc =
+            sscanf(res, "%f#", &mpsas);
+        if (rc < 1)
+        {
+            LOGF_ERROR("Failed to parse input %s", res);
+            return false;
+        } else 
+        {
+            if (fp != nullptr) {
+                LOG_DEBUG("save reading...");
+                fprintf(fp, "%f\t%s\n", mpsas, timestamp());
+                fflush(fp);
+            }
+            count++;
+        }
+
+        AverageReadingN[0].value = mpsas;
+        sumSQ += mpsas;
+
+        if (count > 1)
+        {
+            AverageReadingN[1].value = sumSQ / count;
+            if (mpsas < AverageReadingN[2].value) {
+                AverageReadingN[2].value = mpsas;
+            }
+            if (mpsas > AverageReadingN[3].value) {
+                AverageReadingN[3].value = mpsas;
+            }
+        } else {
+            AverageReadingN[2].value = mpsas;
+            AverageReadingN[3].value = mpsas;
+        }
+        //NELM (see http://unihedron.com/projects/darksky/NELM2BCalc.html)
+        AverageReadingN[4].value = 7.93-5*log(pow(10,(4.316-(mpsas/5)))+1)/log(10);
+    }
+    return true;
+}
+
+const char *LPM::getDefaultName()
+{
+    return "Astromechanics LPM";
```

Comment:
The same name should be used in the XML name attribute of the driver element.. so I suggest you update the XML file to:

```
<device label="Astromechanics LPM">
                <driver name="Astromechanics LPM">indi_astromech_lpm</driver>
                <version>0.1</version>
</device>
```